### PR TITLE
Update 23/05/2018: Escape string throught Array like mylsq base lib

### PR DESCRIPTION
**Pull request by seventrust**
This change will allow insert / replace / update querys with escape strings as in the mysql library

### CODE REVIEW

**Comment by daviemakz** [comment]
Nice work, but the tests are failing so I wont be able to merge this into the source. Can you resolve this, update your pull request and I will have a look again

File: lib/pool.js [hunk 1]
```diff
@@ -215,39 +215,71 @@ class Core extends PoolFunctions {
...
+          } else {
+            return;
+          }
         } else {
-          return;
+          // Release connection
+          scope.connection.release();
+          // Callback [SUCCESS]
+          if (typeof scope.returnCallback === 'function') {
+            scope.returnCallback(rows, {});
```
Comment (MarkHerhold) [inline lib/pool.js:240]: What's the signature of `returnCallback`? Everywhere else the 2nd param appears to be an Error object or string.

Comment (daviemakz) [reply lib/pool.js:240]: Apologies for the late response, what do you mean by signature? The `returnCallback` first parameter should always be an array while the second will be an object (containing error information)

**Comment by daviemakz** [comment]
@seventrust I have updated the tests, please merge with master, commit, push and update your PR. It should kick off another test and hopefully pass! :)

**Comment by seventrust** [comment]
Hi friend! 

let me review a few thing before the PR, in the Query response i see if there an error then the error is an array or an object when the response its successful, this can be better if in both case the error is an object like mysql-js

**Comment by daviemakz** [comment]
I see your point. Essentially since im checking for whether an err object is passed, ive decided to pass the err object as it is generated from the `mysql` module. I think this would be best instead of constructing a custom object.

**Pull request closed without merging**